### PR TITLE
Search Results UI

### PR DESCRIPTION
**Pull request by mingkunm**
### Previews
image
image
image
image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/routes/dsl_route.ts [hunk 1]
```diff
@@ -32,9 +32,13 @@ export function registerDslRoute({ router }: { router: IRouter }) {
...
       } catch (error) {
         if (error.statusCode !== 404) console.error(error);
+
+        // Template: Error: {{Error.type}} – {{Error.reason}}
+        const errorMessage = `Error: ${error.body?.error?.type} - ${error.body?.error?.reason}`;
+
         return response.custom({
           statusCode: error.statusCode || 500,
-          body: error.message,
+          body: errorMessage,
```
Comment (ps48) [inline server/routes/dsl_route.ts:41]: Do we need something similar for `ServiceEndpoints.GetIndexes` endpoint too?

File: public/components/query_compare/search_result/result_components/result_components.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  queryResult2: SearchResults;
+}
+
+const InitialState = () => {
+  return (
+    <EuiPanel
+      hasBorder={false}
+      hasShadow={false}
+      grow={true}
+      style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}
```
Comment (ps48) [inline public/components/query_compare/search_result/result_components/result_components.tsx:25]: Do you want to push this styling to `scss` file?

**Review by msfroh — APPROVED** [review]
Thanks @mingkunm. This is looking really nice!

File: public/components/query_compare/search_result/result_components/result_grid.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+          <span
+            style={{ color: '#69707D', fontSize: '12px', fontWeight: '400', lineHeight: '21px' }}
+          >
+            Not applicable
+          </span>{' '}
+          <EuiIconTip
+            aria-label="IconTip"
+            size="m"
+            type="questionInCircle"
+            color="#343741"
```
Comment (ps48) [inline public/components/query_compare/search_result/result_components/result_grid.tsx:93]: You have hard-coded colors here. Do we know if it works as expected with change of themes in Dashboards. 
Check for theme setting here: https://playground.opensearch.org/app/management/opensearch-dashboards/settings

File: public/components/query_compare/search_result/result_components/result_grid.tsx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  //     setResultGrid(
+  //       queryResult.hits.hits.map((doc: any, id: number) => {
+  //         return (
+  //           <>
+  //             <tr className="osdDocTable__row">{getTds(doc._source)}</tr>
+  //           </>
+  //         );
+  //       })
+  //     );
+  // }, [queryResult]);
```
Comment (ps48) [inline public/components/query_compare/search_result/result_components/result_grid.tsx:209]: Let's remove commented code. If not utilized.

**Pull request merged**